Adds SlotFills/meta box example

mboynes opened the pull request:
This PR adds an example of using the `PluginDocumentSettingPanel` SlotFill to mimic a "legacy" meta box, purely in Gutenberg. This component then uses `useEntityProp()` to read and write post meta.

Here's a screenshot of the rendered meta box:

![Screen Shot 2021-09-30 at 1 56 06 PM](https://user-images.githubusercontent.com/465154/135506614-ff7ac961-4c85-4ef5-ad7c-003245515da6.png)


TODO:
-------

- [ ] Add ES5 implementation

Resolves #12.

mboynes committed:
Adds SlotFills/meta box example

ryanwelcher commented on 07-slotfills-esnext/index.php:
```diff
@@ -0,0 +1,54 @@
+	// automatically load dependencies and version.
+	$asset_file = include(plugin_dir_path(__FILE__) . 'build/index.asset.php');
+
+	wp_register_script(
+		'gutenberg-examples-07-esnext',
+		plugins_url( 'build/index.js', __FILE__ ),
+		$asset_file['dependencies'],
+		$asset_file['version'],
+		true
+	);
+
+	register_block_type(
+		'gutenberg-examples/example-07-esnext',
+		[
+			'editor_script' => 'gutenberg-examples-07-esnext',
+		]
+	);
```
This is a little confusing to me as to why we're calling `register_block_type` when this feature doesn't actually register a block. I understand the use-case ( to enqueue the script ) but we can do this by calling `wp_enqueue_script` instead of `wp_register_script` above. If there are concerns about when loading the script at the correct time, I'd think that the `dependencies` should handle that correctly. Is there another benefit of this approach I am not aware of?

ryanwelcher commented on 07-slotfills-esnext/index.php:
```diff
@@ -0,0 +1,54 @@
+	}
+
+	// automatically load dependencies and version.
+	$asset_file = include(plugin_dir_path(__FILE__) . 'build/index.asset.php');
```
Minor coding standards issue
```suggestion
	$asset_file = include( plugin_dir_path( __FILE__ )  .  'build/index.asset.php' );
```

ryanwelcher commented on 07-slotfills-esnext/src/index.js:
```diff
@@ -0,0 +1,58 @@
+import { TextControl } from '@wordpress/components';
```
Some minor eslint items in this file - there is a .eslintrc file in some of the earlier examples that you can copy into this directory.

ryanwelcher commented on 07-slotfills-esnext/src/index.js:
```diff
@@ -0,0 +1,58 @@
+import { TextControl } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
```
```suggestion
/**
 * WordPress dependencies
 */
import { useEntityProp } from '@wordpress/core-data';
```

ryanwelcher reviewed (changes requested):
Thank you for putting this together! I just had one suggestion and there are some minor phpcs/linting items. I am actively working to improve the tooling here so I am not too concerned with those at the moment.

This is a great addition to the examples, thanks again!

mboynes committed:
Address JS formatting issues

mboynes committed:
Removes unnecessary call to register_block_type

mboynes commented:
Hey @ryanwelcher, thanks for the review! I believe I've addressed everything requested. My only point of confusion was the "WordPress dependencies" docblock. Your suggestion was adding it to line 2, below the `TextControl` import. I couldn't tell if that was a mistake or not, so I added it above that line 🤷 . Regardless, have a look and let me know if you need anything else updated!

ryanwelcher reviewed (approved):
Thanks for the updates! Code looks 👍 and works as expected.
